mediapipe/calculators/ovms/modelapiovmsinferencecalculator.cc

```diff
@@ -93,10 +159,98 @@ static Tensor convertOVTensor2MPTensor(const ov::Tensor& t) {
+        /*
+    undefined,
+    dynamic,
+*/
+    };
+    auto it = precisionMap.find(type);
+    if (it == precisionMap.end()) {
+        return Precision::UNDEFINED;
+    }
+    return it->second;
+}
+
+ov::element::Type_t ovmsPrecisionToIE2Precision(Precision precision) {
+    static std::unordered_map<Precision, ov::element::Type_t> precisionMap{
+        {Precision::FP64, ov::element::Type_t::f64},
+        {Precision::FP32, ov::element::Type_t::f32},
+        {Precision::FP16, ov::element::Type_t::f16},
+        {Precision::I64, ov::element::Type_t::i64},
+        {Precision::I32, ov::element::Type_t::i32},
+        {Precision::I16, ov::element::Type_t::i16},
+        {Precision::I8, ov::element::Type_t::i8},
+        {Precision::I4, ov::element::Type_t::i4},
+        {Precision::U64, ov::element::Type_t::u64},
+        {Precision::U32, ov::element::Type_t::u32},
+        {Precision::U16, ov::element::Type_t::u16},
+        {Precision::U8, ov::element::Type_t::u8},
+        {Precision::U4, ov::element::Type_t::u4},
+        {Precision::U1, ov::element::Type_t::u1},
+        {Precision::BOOL, ov::element::Type_t::boolean},
+        {Precision::BF16, ov::element::Type_t::bf16},
+        {Precision::UNDEFINED, ov::element::Type_t::undefined},
+        {Precision::DYNAMIC, ov::element::Type_t::dynamic}
+        //    {Precision::MIXED, ov::element::Type_t::MIXED},
+        //    {Precision::Q78, ov::element::Type_t::Q78},
+        //    {Precision::BIN, ov::element::Type_t::BIN},
+        //    {Precision::CUSTOM, ov::element::Type_t::CUSTOM
+    };
+    auto it = precisionMap.find(precision);
+    if (it == precisionMap.end()) {
+        return ov::element::Type_t::undefined;
+    }
+    return it->second;
+}
+
+Precision TFSPrecisionToOvmsPrecision(const TFSDataType& datatype) {
+    static std::unordered_map<TFSDataType, Precision> precisionMap{
+        {TFSDataType::DT_FLOAT, Precision::FP32},
+        {TFSDataType::DT_DOUBLE, Precision::FP64},
+        {TFSDataType::DT_HALF, Precision::FP16},
+        {TFSDataType::DT_INT64, Precision::I64},
+        {TFSDataType::DT_INT32, Precision::I32},
+        {TFSDataType::DT_INT16, Precision::I16},
+        {TFSDataType::DT_INT8, Precision::I8},
+        {TFSDataType::DT_UINT64, Precision::U64},
+        {TFSDataType::DT_UINT16, Precision::U16},
+        {TFSDataType::DT_UINT8, Precision::U8},
+        {TFSDataType::DT_BOOL, Precision::BOOL}};
+    auto it = precisionMap.find(datatype);
+    if (it == precisionMap.end()) {
+        return Precision::UNDEFINED;
+    }
+    return it->second;
+}
+
 static tensorflow::Tensor convertOVTensor2TFTensor(const ov::Tensor& t) {
     using tensorflow::Tensor;
     using tensorflow::TensorShape;
-    auto datatype = ovms::getPrecisionAsDataType(ovms::ovElementTypeToOvmsPrecision(t.get_element_type()));
+    auto datatype = getPrecisionAsDataType(ovElementTypeToOvmsPrecision(t.get_element_type()));
```

Comment:
Here don't do two step conversion - add OV datatype to TF datatype

Reply:
Done.